exporter/sapmexporter/exporter.go

```diff
@@ -126,17 +124,26 @@ func (se *sapmExporter) pushTraceData(ctx context.Context, td ptrace.Traces) err
 	return nil
 }
 
-func (se *sapmExporter) retrieveAccessToken(md ptrace.ResourceSpans) string {
+func (se *sapmExporter) retrieveAccessToken(ctx context.Context, md ptrace.ResourceSpans) string {
 	if !se.config.AccessTokenPassthrough {
 		// Nothing to do if token is pass through not configured or resource is nil.
 		return ""
 	}
 
-	attrs := md.Resource().Attributes()
-	if accessToken, ok := attrs.Get(splunk.SFxAccessTokenLabel); ok {
-		return accessToken.Str()
+	var token string
+	switch {
+	case ctx.Value(splunk.SFxAccessTokenHeader) != nil:
```

Comment:
The linter error made me realize that this is not going to work. You need to use https://github.com/open-telemetry/opentelemetry-collector/blob/main/client/client.go#L127

Reply:
Look at the header setter extension https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/a4405af545b3c169cbd5251705ce13c34dd4cbe8/extension/headerssetterextension/internal/source/context.go#L19